### DISCUSSION TITLE
Update SiteSettings in a job

### DISCUSSION THREAD

**Discussion by intellimedhu** [OrchardCMS/OrchardCore#7723] [Q&A]
`SiteSettings` update doesn't take effect if it's set in a background task (or through an API request). Is it a bug or should I store my custom settings in another record?

**Comment by deanmarcussen** [comment]
Can you share your code. It's possible you are using the incorrect methods for updating.

**Comment by intellimedhu** [comment]
```
public async Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToken cancellationToken)
{
    var siteService = serviceProvider.GetRequiredService<ISiteService>();
    var site = await siteService.LoadSiteSettingsAsync();
    site.Alter<MySettings>(nameof(MySettings), mySettings =>
    {
        mySettings.IntProperty += 3;
    });
    await siteService.UpdateSiteSettingsAsync(site);
}
```

update info: Accessing the tenant's scope from another tenant the original tenant's settings are overriden. 

Please, view readme and code about the issue here: https://github.com/intellimedhu/SiteSettings-issue-between-tenants-in-OrchardCore

**Comment by intellimedhu** [comment]
Any ideas? @deanmarcussen

**Reply by deanmarcussen** [reply]
It feels like from what you are describing that the update does work, but you are suggesting that you're getting cross over between tenants?

Can you confirm?

Also I looked at your code and I think you're running RC2.

It might be worth updating to the current dev feed and seeing if the issue still presents the same way.

If it does, can you create an actual issue, as we'll need to take a closer look to understand.

**Reply by intellimedhu** [reply]
Ok, solution is now upgraded to .NET 5 and OrchardCore 15552

Yes, i'm getting cross over between tenants: trying to get Tenant1's `SiteSettings` from Tenant2.

After the upgrade:
Updating `SiteSettings` from the job works as expected. However its not possible to get `ISiteSettings` service of Tenant1 from Tenant2. In my repo see: `TestController` at line 27 `ISiteService` is resolved from Tenant2 instead of Tenant1.

**Reply by jtkech** [reply]
@intellimedhu 

            var scope = await _shellHost.GetScopeAsync("Tenant1");
            await scope.UsingAsync(async scope =>
            {
                var siteService = scope.ServiceProvider.GetRequiredService<ISiteService>();
                var site = await siteService.GetSiteSettingsAsync();
                // ...
            });

**Reply by intellimedhu** [reply]
Thanks, now looks everything working fine.
I'll upgrade my major project and test these options.
